src/extension.ts

```diff
@@ -1,27 +1,153 @@
 // The module 'vscode' contains the VS Code extensibility API
 // Import the module and reference it with the alias vscode in your code below
 import * as vscode from 'vscode';
+import axios from 'axios';
 
 // this method is called when your extension is activated
 // your extension is activated the very first time the command is executed
 export function activate(context: vscode.ExtensionContext) {
+    vscode.window.showInformationMessage('out here!');
 
-	// Use the console to output diagnostic information (console.log) and errors (console.error)
-	// This line of code will only be executed once when your extension is activated
-	console.log('Congratulations, your extension "reactdocs" is now active!');
+    const provider = new DocumentationViewProvider();
+    let viewType = "reactdocs.documentation";
+    let disposable = vscode.window.registerWebviewViewProvider(viewType, provider);
+    context.subscriptions.push(disposable);
+}
+
+interface ReferenceResponse {
+    success: boolean;
+    data: Component;
+}
+interface Component {
+    name: string;
+    link: string;
+    docstring: string;
+}
+
+class DocumentationViewProvider implements vscode.WebviewViewProvider{
+    private disposables: vscode.Disposable[] = [];
+    private view?: vscode.WebviewView;
+    private viewType: string = "reactdocs.documentation";
+
+    constructor() {
+        vscode.window.onDidChangeActiveTextEditor(() => {
+            this.update();
+        }, null, this.disposables)
+        vscode.window.onDidChangeTextEditorSelection(() => {
+            this.update();
+        }, null, this.disposables)
+    }
 
-	// The command has been defined in the package.json file
-	// Now provide the implementation of the command with registerCommand
-	// The commandId parameter must match the command field in package.json
-	let disposable = vscode.commands.registerCommand('reactdocs.helloWorld', () => {
-		// The code you place here will be executed every time your command is executed
+    dispose(){
+        // may need to implement
+    }
 
-		// Display a message box to the user
-		vscode.window.showInformationMessage('Hello World from reactdocs!');
-	});
+    // Called when a view first becomes visible
+    // (happens when view is first loaded/user hides and then shows the view again)
+    resolveWebviewView(
+        webviewView: vscode.WebviewView, 
+        context: vscode.WebviewViewResolveContext, 
+        token: vscode.CancellationToken
+    ){
+        webviewView.webview.options = {
+			enableScripts: true
```

Comment:
change tabs to spaces !! also left indent this line so it is only 1 indent in from the line above